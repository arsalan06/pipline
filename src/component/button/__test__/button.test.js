import React from "react";
import ReactDOM from "react-dom";
import Button from "../Button";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
// afterEach(cleanup);
it("renders without crashing", () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  //   const div = document.createElement("div");
  root.render(<Button />);
});
it("render button correctly", () => {
  render(<Button label="Click me" />);
  const but = screen.getByTestId("button");
  expect(but).toBeInTheDocument();
});
it("changes the class when hovered", () => {
  const component = renderer.create(<Button label="save" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
