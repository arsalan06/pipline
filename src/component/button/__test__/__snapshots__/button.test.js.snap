// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  save
</div>
`;
